Dockerfile

```diff
@@ -85,7 +85,7 @@ RUN set -euo pipefail && \
     pushd "${POETRY_SYSTEM_PROJECT_DIR}"; \
     if [[ "${AIRFLOW_NORM_VERSION}" == "2.1.*" ]]; then \
         poetry add \
-            "apache-airflow[statsd]==${AIRFLOW_NORM_VERSION}" \
+            "apache-airflow==${AIRFLOW_NORM_VERSION}" \
```

Comment:
Any extra packages should be added subsequently as 
`poetry add apache-airflow-providers-apache-slack` or `poetry add apache-airflow-providers-amazon`

Full list here:
https://airflow.apache.org/docs/apache-airflow/stable/extra-packages-ref.html


Note that the extra packages follow their own versioning, e.g.
`poetry add apache-airflow-providers-amazon==1.4.0` at the point of time that the latest release of Airflow is `apache-airflow==2.1.0`

Reply:
Poetry doesn't seem to resolve the dependencies properly when extra packages are specified like so `apache-airflow[apache.spark]`, with the following error being observed

```
#7 15.39 Updating dependencies
#7 15.39 Resolving dependencies...
#7 165.2
#7 165.2   SolverProblemError
#7 165.2
#7 165.2   Because apache-airflow (2.1.0) depends on apache-airflow (2.1.0) which depends on cattrs (>=1.1,<2.0), apache-airflow (2.1.0) requires cattrs (>=1.1,<2.0).
#7 165.2   And because apache-airflow (2.1.0) depends on cattrs (>=1.0,<1.1.0), apache-airflow is forbidden.
#7 165.2   So, because no versions of apache-airflow match >2.1.0,<2.2.0
#7 165.2    and system depends on apache-airflow (2.1.*), version solving failed.
#7 165.2
#7 165.2   at /opt/poetry/lib/poetry/puzzle/solver.py:241 in _solve
#7 165.2       237│             packages = result.packages
#7 165.2       238│         except OverrideNeeded as e:
#7 165.2       239│             return self.solve_in_compatibility_mode(e.overrides, use_latest=use_latest)
#7 165.2       240│         except SolveFailure as e:
#7 165.2     → 241│             raise SolverProblemError(e)
#7 165.2       242│
#7 165.2       243│         results = dict(
#7 165.2       244│             depth_first_search(
#7 165.2       245│                 PackageNode(self._package, packages), aggregate_package_nodes 
```

Reply:
Does it work if the extra packages are placed together in the same command of `poetry add`?

Reply:
Yes it does. Rephrased my above comment to clarify what is happening exactly.